src/client-functions/selectors/add-api.js

```diff
@@ -118,6 +118,24 @@ function createPrimitiveGetterWrapper (observedCallsites, callsite) {
     };
 }
 
+function checkForExcessiveAwaits (snapshotPropertyCallsites, callsiteToCheck) {
+    const key = `${callsiteToCheck.filename}:${callsiteToCheck.lineNum}`;
+
+    let callsitesToAssert = [ callsiteToCheck ];
+
+    // NOTE: If the calliste already exists, but is not asserted, it means that there are
+    // multiple awaited callsites in one assertion. We raise a warning for each of them.
+    if (snapshotPropertyCallsites[key] && snapshotPropertyCallsites[key].asserted === false)
```

Comment:
Rewrite using the following structure
```js
if ()
else if ()
else
```

Reply:
Adding another callsite  to the `callsitesToAssert`
`callsitesToAssert = snapshotPropertyCallsites[key].callsites.concat(callsitesToAssert);`

and setting `snapshotPropertyCallsites[key]` 
`snapshotPropertyCallsites[key] = { callsites: callsitesToAssert, checked: false };`

are not mutually exclusive - we still want to set `snapshotPropertyCallsites[key]`, even if we added a callsite to the list.

At the same time, setting `snapshotPropertyCallsites[key]` and deleting it are mutually exclusive - we don't want to do both.

I've rewrote these conditions as you suggested, but I had to use this line twice:
`snapshotPropertyCallsites[callsiteId] = { callsites: callsitesToAssert, checked: false };`

Reply:
Actually, nevermind. I think, I can simplify it a little differently